src/ray/core_worker/object_interface.cc

```diff
@@ -3,18 +3,38 @@
 #include "ray/common/ray_config.h"
 #include "ray/core_worker/object_interface.h"
 #include "ray/core_worker/store_provider/local_plasma_provider.h"
+#include "ray/core_worker/store_provider/memory_store_provider.h"
 #include "ray/core_worker/store_provider/plasma_store_provider.h"
 
 namespace ray {
 
+void GetObjectIdsPerStoreProvider(
+    const std::vector<ObjectID> &object_ids,
+    EnumUnorderedMap<StoreProviderType, std::unordered_set<ObjectID>> *results) {
+  for (const auto &object_id : object_ids) {
+    // By default use `PLASMA` store provider.
+    auto type = StoreProviderType::PLASMA;
+    if (object_id.IsReturnObject() &&
+        object_id.GetTransportType() ==
```

Comment:
Unless we're using `GetTransportType()` in other contexts, I would prefer just having a `IsDirectActorObject()` method.

Reply:
`GetTransportType()` is an existing method of `ObjectID` and is not added by this PR. I'm thinking the check for `IsReturnObject() && IsReturnObject()` is clear enough here, thus I would prefer not to have another function to wrap this two. Let me know if this makes sense to you.